c2corg_api/views/user_mailinglists.py

```diff
@@ -0,0 +1,82 @@
+from c2corg_api import DBSession
+from c2corg_api.models.user import User
+from c2corg_api.models.mailinglist import Mailinglist
+from c2corg_api.views import cors_policy, restricted_json_view, \
+    restricted_view
+from c2corg_common.attributes import mailinglists
+from cornice.resource import resource
+
+
+def validate_mailinglist_statuses(request, **kwargs):
+    tested = []
+    for ml in request.validated:
+        if ml not in mailinglists:
+            request.errors.add(
+                'body', ml, 'Mailing list `{0}` does not exist'.format(ml))
+        if type(request.validated[ml]) != bool:
+            request.errors.add(
+                'body', ml,
+                'Status `{0}` of mailing list `{1}` should be boolean'.format(
+                    request.validated[ml], ml))
+        if ml in tested:
+            request.errors.add(
+                'body', ml, 'Duplicated mailing list status for `{0}`'.format(
+                    ml))
+        tested.append(ml)
+
+
+@resource(path='/users/mailinglists', cors_policy=cors_policy)
+class UserMailinglistsRest(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @restricted_view()
+    def get(self):
+        """Get the mailinglists subscriptions of the authenticated user.
+
+        Request:
+            `GET` `/users/mailinglists`
+
+        Example response:
+
+            {'avalanche': False, 'lawinen': True, 'valanghe': False, ...}
+
+        """
+        user_id = self.request.authenticated_userid
+        listnames = DBSession.query(Mailinglist.listname). \
+            filter(Mailinglist.user_id == user_id).all()
+
+        return {ml: ml in mailinglists for ml in listnames}
```

Comment:
OK thanks for pointing the problem. I was unsure where this ``nose.proxy.TypeError: keys must be a string`` error came from.